precompiles/collective/src/lib.rs

```diff
@@ -0,0 +1,281 @@
+		let proposal_length: u32 = proposal.len().try_into().map_err(|_| {
+			RevertReason::value_is_too_large("uint32")
+				.in_field("length")
+				.in_field("proposal")
+		})?;
+
+		let proposal_index = pallet_collective::Pallet::<Runtime, Instance>::proposal_count();
+		let proposal_hash: H256 = Runtime::Hashing::hash(&proposal).into();
+		let proposal = Runtime::Call::decode(&mut &*proposal)
+			.map_err(|_| RevertReason::custom("Failed to decode proposal").in_field("proposal"))?
+			.into();
+		let proposal = Box::new(proposal);
+
+		{
+			let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+			RuntimeHelper::<Runtime>::try_dispatch(
+				handle,
+				Some(origin).into(),
+				pallet_collective::Call::<Runtime, Instance>::propose {
+					threshold,
+					proposal,
+					length_bound: proposal_length,
+				},
+			)?;
+		}
+
+		// In pallet_collective a threshold < 2 means the proposal has been
+		// executed directly.
+		let log = if threshold < 2 {
+			log2(
+				handle.context().address,
+				SELECTOR_LOG_EXECUTED,
+				proposal_hash,
+				Vec::new(),
+			)
+		} else {
+			log4(
+				handle.context().address,
+				SELECTOR_LOG_PROPOSED,
+				handle.context().caller,
+				H256::from_slice(&EvmDataWriter::new().write(proposal_index).build()),
+				proposal_hash,
+				EvmDataWriter::new().write(threshold).build(),
+			)
+		};
+
+		handle.record_log_costs(&[&log])?;
+		log.record(handle)?;
+
+		Ok(succeed(&[]))
+	}
+
+	fn vote(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		read_args!(handle, {
+			proposal_hash: H256,
+			proposal_index: u32,
+			approve: bool
+		});
+
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+		RuntimeHelper::<Runtime>::try_dispatch(
+			handle,
+			Some(origin).into(),
+			pallet_collective::Call::<Runtime, Instance>::vote {
+				proposal: proposal_hash.into(),
+				index: proposal_index,
+				approve,
+			},
+		)?;
+
+		// TODO: Since we cannot access ayes/nays of a proposal we cannot
+		// include it in the EVM events to mirror Substrate events.
+
+		let log = log3(
+			handle.context().address,
+			SELECTOR_LOG_VOTED,
+			handle.context().caller,
+			proposal_hash,
+			EvmDataWriter::new().write(approve).build(),
+		);
+		handle.record_log_costs(&[&log])?;
+		log.record(handle)?;
+
+		Ok(succeed(&[]))
+	}
+
+	fn close(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		read_args!(handle, {
+			proposal_hash: H256,
+			proposal_index: u32,
+			proposal_weight_bound: u64,
+			length_bound: u32
+		});
+
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+		RuntimeHelper::<Runtime>::try_dispatch(
+			handle,
+			Some(origin).into(),
+			pallet_collective::Call::<Runtime, Instance>::close {
+				proposal_hash: proposal_hash.into(),
+				index: proposal_index,
+				proposal_weight_bound,
+				length_bound,
+			},
+		)?;
+
+		// TODO: How to know if the proposal was executed or not?
```

Comment:
We could know whether it was approved/disapproved by the weight info or pays fees Yes/No, although it might be overkill..